WordPress/gutenberg
# [LinkControl] Extract reusable parts

## Description
This PR refactors a monolithic LinkControl into separate, bite-sized components and hooks that are reusable. This PR paves ground for reusing the `LinkControlSearchInput` component in the following interaction designed as a part of https://github.com/WordPress/gutenberg/issues/23375:

<img width="358" alt="85623474-75db0a80-b636-11ea-9348-dffea4d2ac71" src="https://user-images.githubusercontent.com/205419/87160057-a374b480-c2c2-11ea-9951-e8951828e881.png">

## How has this been tested?
1. Create a new post
1. Add navigation block, create a few links
1. Click the edit link button in the toolbar
1. Confirm the behavior with and without this PR is exactly the same

## Screenshots

<img width="553" alt="Zrzut ekranu 2020-07-10 o 15 38 34" src="https://user-images.githubusercontent.com/205419/87160568-72e14a80-c2c3-11ea-9184-1850a32505bd.png">

## Types of changes
Non-breaking change
